File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> nums_map = {};

        for (int i=0;i<nums.size(); ++i) {
            nums_map[nums[i]] = i;
        }

        for (int i=0;i<nums.size(); ++i) {
            int pot = target - nums[i];
            if (nums_map.find(pot) != nums_map.end() && nums_map[pot]!=i) {
                return vector<int>{min(nums_map[pot], i), max(nums_map[pot], i)};
            }
        }


    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n = len(nums)
        l = 0
        r = n-1
        m = (r-l)//2

        while l < r:
            a = nums[l]
            b = nums[m]
            c = nums[r]

            if r-l == 1:
                return min(a,c)

            if c < b:
                l = m
            else:
                r = m
            m = l + (r-l)//2

        return  nums[l]
                
